app/controllers/todo_lists_controller.rb

```diff
@@ -0,0 +1,61 @@
+class TodoListsController < ApplicationController
+    # GET /users/:user_id/todo_lists
+    def index
+        @user = User.find_by(id: params[:user_id]) # Fetch user by ID
+        if @user
+          @todo_lists = @user.todo_lists
+          render json: @todo_lists
+        else
+          render json: { error: "User not found" }, status: :not_found
+        end
+      end
+      
+  
+    # POST /todo_lists
+    def create
+      user = User.find_by(id: todo_list_params[:user_id])
+      if user
+        todo_list = user.todo_lists.new(title: todo_list_params[:title])
+        if todo_list.save
+          render json: todo_list, status: :created
+        else
+          render json: { errors: todo_list.errors.full_messages }, status: :unprocessable_entity
+        end
+      else
+        render json: { error: "User not found" }, status: :not_found
+      end
```

Comment:
Too many if else conditions. Try to handle it with some generalised function for repeated cases.